sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java

```diff
@@ -42,15 +42,13 @@
   // Array for each type. Only 1 is populated for any type.
   private byte[] byteData;
   private short[] shortData;
+  // This is not only used to store data for int column vector, but also can store offsets and
+  // lengths for array column vector.
   private int[] intData;
```

Comment:
Good catch. Is it possible to use `longData`, which has a pair of 32-bit offset and length, to keep `MAX_CAPACITY` array length?